lang/wasm.cc

```diff
@@ -290,51 +197,60 @@ struct from_val_visitor {
             "f32",
             "f64",
             "v128",
-            "externref",
             "funcref",
+            "externref",
         };
         if (val.kind() != expected) {
             throw wasm::exception(format("Incorrect wasm value kind returned. Expected {}, got {}", kind_str[size_t(expected)], kind_str[size_t(val.kind())]));
         }
     }
 };
 
+void compile(context& ctx, const std::vector<sstring>& arg_names, std::string script) {
+    try {
+        ctx.module = wasmtime::create_module(ctx.engine_ptr, rust::Str(script.data(), script.size()));
+        auto store = wasmtime::create_store(ctx.engine_ptr);
+        auto inst = create_instance(ctx.engine_ptr, **ctx.module, *store);
```

Comment:
Shouldn't creation of an module and store/instance be separated? We create many instances per module.

Reply:
The module is indeed created only once. In this case we create an instance (and a Func) alongside it to confirm that the UDF is usable (the compilation of the module can succeed for basically any valid WASM code, which may or may not contain our function). 
After confirming this, the instance is destroyed and other instances are created only when we need them for UDF executions

Reply:
Ok, please add a comment to explain.

Reply:
Comment added